monkey/infection_monkey/network/firewall.py

```diff
@@ -44,19 +49,23 @@ def __init__(self):
 
     def is_enabled(self):
         try:
-            cmd = subprocess.Popen("netsh advfirewall show currentprofile", stdout=subprocess.PIPE)
-            out = cmd.stdout.readlines()
-
-            for line in out:
-                if line.startswith("State"):
-                    state = line.split()[-1].strip()
-
-            return state == "ON"
+            out = subprocess.check_output(
+                "netsh advfirewall show currentprofile", timeout=SHORT_REQUEST_TIMEOUT
+            )
+        except subprocess.TimeoutExpired:
+            return None
         except Exception:
             return None
 
+        for line in out.decode().splitlines():
+            if line.startswith("State"):
+                state = line.split()[-1].strip()
+                return state == "ON"
+
+        return None
```

Comment:
This function is named `is_enabled()`. Should it be returning `bool` instead of `Optional[str]`?

Reply:
It is returning `Optional[bool]`

Reply:
That's what I mean. Why optional? Why not just return bool?